app/src/main/kotlin/com/wire/android/ui/home/conversations/messages/ConversationMessagesViewModel.kt

```diff
@@ -135,7 +143,7 @@ class ConversationMessagesViewModel @Inject constructor(
         )
 
         if (resultData != null) {
-            showOnAssetDownloadedDialog(assetName, resultData, assetSize, messageId)
+            showOnAssetDownloadedDialog(assetContent.name ?: "", resultData, assetContent.sizeInBytes, messageId)
```

Comment:
Asset name should never be empty or null.

Reply:
When it was a `UIMessageContent`, this was the case.

`MessageContentMapper.kt`
```kotlin
                // It's a generic Asset Message so let's not download it yet
                else -> {
                    UIMessageContent.AssetMessage(
                        assetName = name ?: "",
                        assetExtension = name?.split(".")?.last() ?: "",
                        assetId = AssetId(remoteData.assetId, remoteData.assetDomain.orEmpty()),
                        assetSizeInBytes = sizeInBytes,
                        downloadStatus = downloadStatus
                    )
```


I'm keeping the same behaviour. The difference is that we don't have a `UIMessageContent`, and we're using `MessageContent` directly.

How should we handle this?

Reply:
Nevermind. I just realised this comes imposed by Kalium restrictions. Maybe we should assess in the not so distant future whether it makes sense to force the asset name to be a not nullable field on the `MessageAssetContent` DB table. 
In the meantime disregard my previous comment 😅